### PR TITLE
MYST-515 / proposal fetcher

### PR DESCRIPTION
**Pull request by ignasbernotas**

### CODE REVIEW

File: src/app/data-fetchers/proposal-fetcher.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  api: TequilApi
+  proposals: Array<ProposalDto> = []
+  loop: FunctionLooper
+  subscribers: Array<Function> = []
+
+  constructor (api: TequilApi) {
+    this.api = api
+  }
+
+  run (interval: number = 5000): this {
```
Comment (Waldz) [inline src/app/data-fetchers/proposal-fetcher.js]: I would put this configuration to constructor

**Comment by Waldz** [comment]
@ignasbernotas, You should rebase your branch onto `feature/MYST-508-tequilapi-with-flow`
```
git branch -u origin/feature/MYST-508-tequilapi-with-flow
git pull -r
```

File: src/app/data-fetchers/proposal-fetcher.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    this.loop = new FunctionLooper(async () => {
```
Comment (Waldz) [inline src/app/data-fetchers/proposal-fetcher.js]: `this.loop` can be initialised in constructor
And `this.start()` just starts the loop

File: test/unit/specs/data-fetchers/proposal-fetcher.spec.js [hunk 3]
```diff
@@ -0,0 +1,96 @@
+/* eslint no-unused-expressions: 0 */
+import {describe, it, expect} from '../../../helpers/dependencies'
+import lolex from 'lolex'
+import ProposalFetcher from '../../../../src/app/data-fetchers/proposal-fetcher'
+import FakeAdapter from '../../../../src/libraries/api/client/adapters/fake-adapter'
+import TequilApi from '../../../../src/libraries/api/client/tequil-api'
+import utils from '../../../helpers/utils'
+
+describe('DataFetchers', () => {
```
Comment (Waldz) [inline test/unit/specs/data-fetchers/proposal-fetcher.spec.js]: Make `tests` structure same as in `src`

File: test/unit/specs/data-fetchers/proposal-fetcher.spec.js [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+        const fetcher = new ProposalFetcher(new TequilApi(adapter))
+        let proposals = []
+
+        fetcher.subscribe((fetchedProposals) => {
+          proposals = fetchedProposals
+        })
+
+        fetcher.run(1001)
+
+        await tickWithDelay(1000)
```
Comment (Waldz) [inline test/unit/specs/data-fetchers/proposal-fetcher.spec.js]: Is it possible to You use `done()` and dont sleep in tests?

Comment (ignasbernotas) [reply test/unit/specs/data-fetchers/proposal-fetcher.spec.js]: Not really, this fetcher depends on donce's FunctionLooper, which uses timeouts.

Comment (ignasbernotas) [reply test/unit/specs/data-fetchers/proposal-fetcher.spec.js]: Plus it doesn't sleep, lolex package allows us to fake the intervals

File: src/main/index.js [hunk 5]
```diff
@@ -18,13 +19,15 @@ global.__sentryURL = SENTRY_CONFIG.publicURL
 global.__static = mysterionConfig.staticDirectoryPath
 
 const tequilApi = dependencies.get('tequilapi')
+const proposalFetcher = new ProposalFetcher(tequilApi)
```
Comment (Waldz) [inline src/main/index.js]: Could You register `proposalFetcher` to DI, please ;)

**Comment by Waldz** [comment]
Rebased onto branch `feature/MYST-508-tequilapi-with-flow`

**Pull request merged**